bindings/python/scripts/parse.py

```diff
@@ -208,7 +180,11 @@ def parse_file(source, compilation_database_path=None):
     )
 
     # Parse the given source code file by running clang and generating the AST before loading
-    source_ast = index.parse(path=source, args=compilation_commands)
+    source_ast = index.parse(
+        path=source,
+        args=compilation_commands,
+        options=clang.TranslationUnit.PARSE_DETAILED_PROCESSING_RECORD,
```

Comment:
Why this option?

Reply:
from #17 seems like it would add the `INCLUSION_DIRECTIVE`-values for `cursor.kind.name`, but then a single-shot PR to solve #17 completely would've made more sense

Reply:
@divmadan Please document the reason in code comments

Reply:
Required to get the `INCLUSION_DIRECTIVE`s (included filenames in the source file) Without this option, clang doesn't retain that information.

From the C API:
```
/**                                                                         
 * \brief Used to indicate that the parser should construct a "detailed"    
 * preprocessing record, including all macro definitions and instantiations.
 *                                                                          
 * Constructing a detailed preprocessing record requires more memory        
 * and time to parse, since the information contained in the record         
 * is usually not retained. However, it can be useful for                   
 * applications that require more detailed information about the            
 * behavior of the preprocessor.                                            
 */                                                                         
CXTranslationUnit_DetailedPreprocessingRecord = 0x01,
```

***

> document the reason in code comments

Done :+1: